### ISSUE TITLE
New linter: preserve caught exceptions as ex-info causes

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2969]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

A catch clause that replaces a caught exception with a new `ex-info` value without passing the original cause loses stack and causal information.

```clojure
(try
  (work)
  (catch Exception e
    (throw (ex-info "Failed" {}))))
```

**Describe the solution you'd like**

Add a `:missing-caught-exception-cause` linter, enabled at warning level by default, when the last catch expression throws a core `ex-info` call with only message and data while the caught binding is named. Do not warn when a cause is supplied, the binding is `_`, or `ex-info` resolves to a local replacement.

**Describe alternatives you've considered**

Unused-binding can sometimes reveal the lost exception, but it does not explain the causal-chain problem and can be suppressed independently. Warning on every new exception in a catch would be too broad.

**Additional context**

I am interested in submitting a focused PR with resolution, ignored-binding, and existing-cause tests if this approach is accepted.

**Comment by NoahTheDuke** [comment]
How often does this happen vs wanting to actually replace the existing throwable?

**Comment by marco-carvalho** [comment]
The pinned Metabase corpus has 11 occurrences, so the pattern is not rare there. That does not establish the intent ratio, though. The current check excludes an ignored `_` binding and calls that already pass a cause, but a named exception can still be logged or inspected before being intentionally replaced. Static analysis cannot reliably distinguish that intent. The corpus result therefore supports an opt-in warning more strongly than a default-on warning.

**Comment by yuhan0** [comment]
If this is ever implemented I feel that it should default to `:off` - 'losing stack and causal information' is a discretionary concern which some users would not find relevant
